std/typecons.d

```diff
@@ -422,19 +422,31 @@ template Tuple(Specs...)
         U u = U.init;
         T t = u;
     }));
+
     /+ Helper for partial instanciation +/
     template isBuildableFrom(U)
     {
         enum isBuildableFrom(T) = isBuildable!(T, U);
     }
 
+    /+ Determine whether or not all given types in T are the same type +/
+    template areAllSame(T...)
+    {
+        static if (T.length == 0 || T.length == 1)
+            enum areAllSame = true;
+        else
+        {
+            enum areAllSame = is(T[0] == T[1]) && areAllSame!(T[1 .. $]);
```

Comment:
We usally try to implement these in binary mode, to limit recursion depth. That said, why are you reimplementing this at all to begin with? Use `allSatisfy`:
```D
    template areAllSame(T...)
    {
		static if (T.length <= 1)
			enum areAllSame = true;
		else
		{
			enum isT(U) = is(U == T[0]);
			enum areAllSame = allSatisfy!(isT, T);
		}
    }
```
With tests: http://dpaste.dzfl.pl/5b09c6c760ed